src/form/input/typeahead/TypeaheadInput.tsx

```diff
@@ -1,19 +1,20 @@
 import React, {useEffect} from "react";
 import classNames from "classnames";
 
-import Input from "../Input";
+import Input, {InputTypes} from "../Input";
 import useDebounce from "../../../core/utils/hooks/debounce";
 
 export interface TypeaheadInputProps {
   onQueryChange: (value: string) => void;
+  name: string;
+  placeholder: string;
+  type?: InputTypes;
```

Comment:
I think the `type` option should only accept searchable types such as`text` and `number`, not all `InputTypes`. It would look really weird if I give `color` as type and there is no real-life scenario to use it.

Reply:
You're right, color type doesn't make sense. We use text type basically. We can remove except `text` and `number` types.

Reply:
Ok. @yasincaliskan let's use `Extract` util on InputTypes to generate a union type with only "text" and "number".